src/main/java/com/dataguise/saas/exception/InternalServerException.java

```diff
@@ -0,0 +1,41 @@
+package com.dataguise.saas.exception;
+
+import org.springframework.http.HttpStatus;
+
+/**
+ * This exceptions are thrown when there is an error in processing from saas-ui backend to frontend.
+ */
+public class InternalServerException extends Exception {
+
+    String errorMessage;
+    HttpStatus status;
+
+    public InternalServerException() {}
+
+    public InternalServerException(String message, HttpStatus status) {
+        super(message);
+        this.errorMessage = message;
+        this.status = status;
+    }
+
+    public InternalServerException(String message) {
+        super(message);
+        this.errorMessage = message;
+    }
+
+    public HttpStatus getStatus() {
+        return status;
+    }
+
+    public void setStatus(HttpStatus status) {
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.design.DesignForExtensionCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/exception/InternalServerException.java L30
    
    Class &#39;InternalServerException&#39; looks like designed for extension (can be subclassed), but the method &#39;setStatus&#39; does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class &#39;InternalServerException&#39; final or making the method &#39;setStatus&#39; static/final/abstract/empty, or adding allowed annotation for the method.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a600775039>*